XenAdmin/Dialogs/LicenseManager/LicenseStatus.cs

```diff
@@ -213,13 +213,12 @@ internal static bool PoolIsMixedFreeAndExpiring(IXenObject xenObject)
                                    into g
                                    select new { ExpiryDate = g.Key, Hosts = g };
 
-                expiryGroups = expiryGroups.OrderBy(g => g.ExpiryDate).ToList();
-
-                if (expiryGroups.Count() <= 1)
-                    return false;
-
-                if ((expiryGroups.ElementAt(1).ExpiryDate - expiryGroups.ElementAt(0).ExpiryDate).TotalDays > 30)
-                    return true;
+                if(expiryGroups.Count() > 1)
+                {
+                    expiryGroups.OrderBy(g => g.ExpiryDate);

```

Comment:
The `OrderBy` function returns the items in sorted order, as far as I am aware it doesn't also do this in place.
Therefore I recommend writing it back into the variable as it used to i.e.:
```csharp
expiryGroups = expiryGroups.OrderBy(g => g.ExpiryDate).ToList();
```